src/Drawing Area/main.vala

```diff
@@ -0,0 +1,31 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+public void main () {
+
+    var drawing_area = (Gtk.DrawingArea) workbench.builder.get_object ("drawing_area");
+    var scale_rotate = (Gtk.Scale) workbench.builder.get_object ("scale");
+
+    double[,] triangle = { { 100.0, 100.0 }, { 0.0, -100.0 }, { -100.0, 100.0 } };
+
+    double angle = 0.0;
+
+    drawing_area.set_draw_func ((area, cr, width, height) => {
```

Comment:
what is `cr`? maybe there is a better name for the variable

Reply:
cr is Cairo context, maybe "context" / "cairo_context" would be a better choice